src/cli/proxy_mode.rs

```diff
@@ -38,12 +38,22 @@ pub fn main() -> Result<()> {
         } else {
             env::args_os().skip(2).collect()
         };
+        let help = cmd_args.is_empty()
+            || cmd_args == &["help"]
+            || cmd_args == &["--help"]
+            || cmd_args == &["-h"];
 
         let cfg = set_globals(false, true)?;
         cfg.check_metadata_version()?;
-        direct_proxy(&cfg, &arg0, toolchain, &cmd_args)?
+        (help, direct_proxy(&cfg, &arg0, toolchain, &cmd_args)?)
     };
 
+    if help {
+        println!(
+            "\nThis proxy of {} supports an additional +toolchain option, see `rustup help`.",
```

Comment:
My suggestion of `rustup proxy` (perhaps `rustup proxy list`) would be a nicer spot for this to reference.  Of course, that doesn't exist yet...